src/csharp/Microsoft.Spark/Utils/UdfSerDe.cs

```diff
@@ -262,18 +262,34 @@ private static TypeData SerializeType(Type type)
                 typeData,
                 td =>
                 {
-                    Type type = AssemblyLoader.LoadAssembly(
+                    Assembly assembly = AssemblyLoader.LoadAssembly(
```

Comment:
I think we can just add the null conditional operator `?.` instead when calling `GetType`.    `AssemblyLoader.LoadAssembly` should already log a warning indicating the assembly filename and the different paths searched.

Reply:
I have switched and it throws the FileNotFound and I can see the warning that logs all of the paths before the exceptions.